### ISSUE TITLE
(aws s3): Can't setup intelligent tiering

### ISSUE THREAD

**Issue by arogozhnikov** [aws/aws-cdk#20937]
### Describe the bug

Below is an intelligent tiering config that I want to add to my setup:

```python
  intelligent_tiering_configurations=[
      s3.IntelligentTieringConfiguration(
          name='intelligent-tiering-all',
      ),
  ],
```

i.e. moving all files to intelligent tiering. 
I don't see a knob that would define which 'tiers' should be used, and likely that's the cause of the next problem.

Here is a diff generated by CDK

```
Stack StaticStack
Resources
[~] AWS::S3::Bucket MyBucketName MyBucketName3AF8577D
 ├─ [+] IntelligentTieringConfigurations
 │   └─ [{"Id":"intelligent-tiering-all","Status":"Enabled","Tierings":[]}]
```

pay attention that Tierings array is empty.

CloudFormation documentation says:

```
Tierings
Specifies a list of S3 Intelligent-Tiering storage class tiers in the configuration. At least one tier must be defined in the list
```

I can't deploy this change, and get the following error

```
The stack named ... failed to deploy: UPDATE_ROLLBACK_COMPLETE: The XML you provided was not well-formed or did not validate against our published schema (Service: Amazon S3; Status Code: 400; Error Code: MalformedXML; Request ID: 4043CP238H4ED1TB; S3 Extended Request ID:  ... ; Proxy: null)
```

### Expected Behavior

Changes should be deployable, and probably tierings should be exposed to the user.

### Current Behavior

Changes can't be deployed when intelligent tiering is  turned on.

### Reproduction Steps

see above

### CDK CLI Version

2.27.0

### Node.js Version

v18.3.0

### OS

Mac OS

### Language

Python

### Language Version

Python 3.9.13

**Comment by peterwoodworth** [comment]
Hey @arogozhnikov,

Tierings will be configured as you supply the number of days you'd like in `archiveAccessTierTime` and `deepArchiveAccessTierTime`. Simply supply your desired duration for each of these properties and the tierings will be filled out

Can make documentation more clear here

**Comment by arogozhnikov** [comment]
point is, I don't want to use archive tiers, instead following explanation by @otaviomacedo from [here](https://github.com/aws/aws-cdk/issues/20783#issuecomment-1171016180)

```
Frequent Access tier (automatic)
Infrequent Access tier (automatic)
Archive Instant Access tier (automatic)
Archive Access tier (optional)
Deep Archive Access tier (optional)
```

I want files to be moved only across first three tiers. By looking at API I assume that not passing `archiveAccessTierTime` and `deepArchiveAccessTierTime` would correspond to using first three tiers only (just as I want them to).

If I'm wrong, what is the way to achieve this behavior in CDK?

**Comment by maxdumas** [comment]
@arogozhnikov just chiming in here: According to the [Cloudformation reference](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-intelligenttieringconfiguration.html) you only need to specify a bucket-level Intelligent-Tiering configuration if you want to use automatic Glacier Archiving or Glacier Deep Archiving (those optional tiers you mentioned). If you're just using the first three automatic tiers of Intelligent Tiering, it suffices to set the per-object storage class to Intelligent-Tiering. If you want the per-object storage clas to be set automatically, you should use a Lifecycle configuration to automatically transition objects to Intelligent-Tiering on creation. The CDK for that should look something like this:

```ts
s3.Bucket(
    self,
    "MyBucket",
    lifecycle_rules=[
        s3.LifecycleRule(
            transitions=[
                s3.Transition(
                    storage_class=s3.StorageClass.INTELLIGENT_TIERING,
                    transition_after=Duration.days(0),
                )
            ]
        )
    ],
)
```

So basically you actually don't want to be using the `intelligent_tiering_configurations` property.

**Comment by arogozhnikov** [comment]
@maxdumas that's what I ended up with, but still wasn't sure about. Thanks for confirming with sources!
